### PR TITLE
feat(hostpreflights): udp port status

### PR DESCRIPTION
**Pull request by emosbaugh**
## Description, Motivation and Context

Adds a collector to check if a UDP port is in use.

Usage example is:

```
apiVersion: troubleshoot.sh/v1beta2
kind: HostPreflight
metadata:
  name: port
spec:
  collectors:
    - udpPortStatus:
        collectorName: flannel
        port: 8472
  analyzers:
    - udpPortStatus:
        collectorName: flannel
        outcomes:
          - fail:
              when: "address-in-use"
              message: Another process was already listening on port 8472.
          - fail:
              when: "error"
              message: Unexpected port status
          - pass:
              when: "connected"
              message: Port 8472 is open
          - warn:
              message: Unexpected port status
```

https://github.com/replicatedhq/troubleshoot.sh/pull/466

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [x] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by emosbaugh** [comment]
[sc-61600]

File: examples/collect/host/udp-port.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: troubleshoot.sh/v1beta2
+kind: HostCollector
+metadata:
+  name: port
+spec:
+  collectors:
+    - tcpPortStatus:
```
Comment (diamonwiggins) [inline examples/collect/host/udp-port.yaml]: guessing this should be udp not tcp

File: examples/collect/remote/udp-port.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: troubleshoot.sh/v1beta2
+kind: HostCollector
+metadata:
+  name: port
+spec:
+  collectors:
+    - tcpPortStatus:
```
Comment (diamonwiggins) [inline examples/collect/remote/udp-port.yaml]: guessing this should be udp not tcp

**Review by diamonwiggins — CHANGES_REQUESTED** [review]
a couple things

**Review by diamonwiggins — APPROVED** [review]
lgtm

**Pull request merged**